cluster-provision/k8s/1.19/istio.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+set -xe
+
+# istio install-cni pods need to run as privileged as they modity /etc/cni/net.d configuration file
+sed -i -r "s#name: install-cni#name: install-cni\n          securityContext:\n            privileged: true#" /var/provision/istio/istio-deployment.yaml
+
+kubectl --kubeconfig /etc/kubernetes/admin.conf create ns istio-system
+kubectl --kubeconfig /etc/kubernetes/admin.conf create -f /var/provision/istio/istio-deployment.yaml
+
+sleep 10
+
+set +e
+
+status=`kubectl --kubeconfig=/etc/kubernetes/admin.conf get pods --all-namespaces --no-headers --field-selector status.phase!=Running --output=name | grep istio`
+retry_counter=0
+while [[ $retry_counter -lt 30 && ! -z "$status" ]]; do
+    sleep 10
+    echo "Waiting for istio pods to be running..."
+    status=`kubectl --kubeconfig=/etc/kubernetes/admin.conf get pods --all-namespaces --no-headers --field-selector status.phase!=Running --output=name | grep istio`
+    retry_counter=$((retry_counter + 1))
+done
```

Comment:
Doing this directly with golang inside gocli is a big no-no right ? It means we have to expose the apiserver port and all.

Reply:
Not sure tbh, I took inspiration from `ceph-csi.sh`. What apiserver port do you mean?